File: my_cpp_pkg/src/robot_news_station.cpp
#include "rclcpp/rclcpp.hpp"
#include "std_msgs/msg/string.hpp"

class RobotNewsStationNode: public rclcpp::Node
{
    public:
        RobotNewsStationNode(): Node("robot_news_station")
        {
            this->declare_parameter("robot_name", "C3P2");

            publisher = this->create_publisher<std_msgs::msg::String>("robot_news", 10);
            robot_name = this->get_parameter("robot_name").as_string();
            counter_ = 0;
            timer = this->create_wall_timer(
                std::chrono::milliseconds(5000),
                std::bind(&RobotNewsStationNode::publishNews, this)
            );
            RCLCPP_INFO(this->get_logger(), "Robot News Station has been started");
        }

    private:
        void publishNews()
        {
            counter_++;
            auto msg = std_msgs::msg::String();
            msg.data = "Hi, this is " + robot_name + " from the Robot News Station - " + std::to_string(counter_);
            publisher->publish(msg);
        }

        std::string robot_name;
        rclcpp::Publisher<std_msgs::msg::String>::SharedPtr publisher;
        rclcpp::TimerBase::SharedPtr timer;
        int counter_;
};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    //auto node = std::make_shared<rclcpp::Node>("cpp_test");
    //RCLCPP_INFO(node->get_logger(), "Hello to my Cpp Node");
    auto node = std::make_shared<RobotNewsStationNode>(); 
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: my_py_pkg/my_py_pkg/robot_news_station.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
#from example_interfaces.msg import String
from std_msgs.msg import String

class RobotNewsStationNode(Node):
    def __init__(self):
        super().__init__("robot_news_station")

        self.declare_parameter("robot_name", "C3P2")

        self.counter = 0
        self.robot_name = self.get_parameter("robot_name").value
        self.publisher = self.create_publisher(String, "robot_news", 10)
        self.timer = self.create_timer(5.0, self.publish_news)
        self.get_logger().info("Robot News Station has been started")

    def publish_news(self):
        self.counter += 1
        msg = String()
        msg.data = "Hi, this is " + str(self.robot_name) + " from the robot news station - " + str(self.counter)
        self.publisher.publish(msg)

def main(args = None):
    rclpy.init(args = args)
    node = RobotNewsStationNode()
    rclpy.spin(node)
    rclpy.shutdown()

if __name__ == "__main__":
    main()